pkg/recipe/variable.go

```diff
@@ -14,14 +14,15 @@ import (
 
 // SystemVariables contain information about a pipeline trigger.
 type SystemVariables struct {
-	PipelineTriggerID  string                 `json:"__PIPELINE_TRIGGER_ID"`
-	PipelineID         string                 `json:"__PIPELINE_ID"`
-	PipelineUID        uuid.UUID              `json:"__PIPELINE_UID"`
-	PipelineReleaseID  string                 `json:"__PIPELINE_RELEASE_ID"`
-	PipelineReleaseUID uuid.UUID              `json:"__PIPELINE_RELEASE_UID"`
-	PipelineOwnerType  resource.NamespaceType `json:"__PIPELINE_OWNER_TYPE"`
-	PipelineOwnerUID   uuid.UUID              `json:"__PIPELINE_OWNER_UID"`
-	ExpiryRuleTag      string                 `json:"__EXPIRY_RULE_TAG"`
+	PipelineTriggerID  string    `json:"__PIPELINE_TRIGGER_ID"`
+	PipelineID         string    `json:"__PIPELINE_ID"`
+	PipelineUID        uuid.UUID `json:"__PIPELINE_UID"`
+	PipelineReleaseID  string    `json:"__PIPELINE_RELEASE_ID"`
+	PipelineReleaseUID uuid.UUID `json:"__PIPELINE_RELEASE_UID"`
+	ExpiryRuleTag      string    `json:"__EXPIRY_RULE_TAG"`
+
+	// PipelineNamespace is the namespace of the requester of the pipeline.
+	PipelineNamespace resource.Namespace `json:"__PIPELINE_NAMESPACE"`
```

Comment:
I agree that if we need to know the UID and owner type of the requester a `resource.Namespace` is better than 2 fields. However, I see 2 problems:
- `PipelineNamespace` isn't self-documented, i.e., if this is used along the code I'll need to go to to the documentation to see whose namespace this is. By default I'd think this is the pipeline owner's namespace, as the resource path for a pipeline is `/namespaces/{namespaceID}/pipelines/{pipelineID}`. I'd use `PipelineRequester` as a name instead.
- This field is redundant with `PipelineRequesterUID`, which should be removed (or at least marked as deprecated for now and cleaned up in a subsequent PR).
```suggestion
	PipelineRequester resource.Namespace `json:"__PIPELINE_REQUESTER"`
```